tests/common/ratelimitpolicy/ratelimitpolicy_controller_test.go

```diff
@@ -407,25 +367,9 @@ var _ = Describe("RateLimitPolicy controller", func() {
 				err = k8sClient.Get(ctx, limitadorKey, existingLimitador)
 				// must exist
 				Expect(err).ToNot(HaveOccurred())
-				Expect(existingLimitador.Spec.Limits).To(ContainElements(limitadorv1alpha1.RateLimit{
-					MaxValue:   1,
-					Seconds:    3 * 60,
-					Namespace:  wasm.LimitsNamespaceFromRLP(rlp),
-					Conditions: []string{fmt.Sprintf(`%s == "1"`, wasm.LimitNameToLimitadorIdentifier(rlpKey, "l1"))},
-					Variables:  []string{},
-					Name:       rlptools.LimitsNameFromRLP(rlp),
-				}))
-			}).WithContext(ctx).Should(Succeed())
-
-			Eventually(func(g Gomega) {
-				// Check gateway back references
-				err = k8sClient.Get(ctx, gwKey, existingGateway)
-				// must exist
-				g.Expect(err).ToNot(HaveOccurred())
-				refs := []client.ObjectKey{rlpKey}
-				serialized, err := json.Marshal(refs)
-				g.Expect(err).ToNot(HaveOccurred())
-				g.Expect(existingGateway.GetAnnotations()).To(HaveKeyWithValue(rlp.BackReferenceAnnotationName(), string(serialized)))
+				Expect(lo.Filter(existingLimitador.Spec.Limits, func(l limitadorv1alpha1.RateLimit, _ int) bool { // a hack to isolate test namespaces sharing the same limitador cr
+					return strings.HasPrefix(l.Namespace, fmt.Sprintf("%s/", testNamespace))
```

Comment:
This is not the right solution. Actually what we want is to no longer create one Kuadrant CR per test suite, but one single instance per cluster only, and write tests with that in mind.

Reply:
I may have misunderstood this before. I thought the multiple test suites were running in parallel to each other, what would cause multiple deployments of the Kuadrant Operator running concurrently in the cluster, as well as multiple Kuadrant CRs, as if these deployments could all be magically isolated from each other.

As @KevFan [explained](https://kubernetes.slack.com/archives/C05J0D0V525/p1729501497688009?thread_ts=1729275538.900229&cid=C05J0D0V525), this is not what happens. The multiple _test suites_ run sequentially to each other. It's only the _test cases_ within a test suite that run in parallel. So we do NOT have multiple Kuadrant Operators running in the cluster at a given time and rarely multiple Kuadrant CRs (with the only exception of specific test cases that may have been thought particularly for this scenario.)

With that now clear, I guess filtering the limits found in the Limitador CR only to the ones that indeed should not be there (i.e. the ones relative to the test case/test namespace in matter) is somewhat acceptable.